generators/infra/templates/httpClient/infra/HttpClientAxios.ts

```diff
@@ -1,30 +1,57 @@
 import cookies from "js-cookie";
-import axios, { AxiosRequestConfig } from "axios";
-import { IHttpClient, HttpMethod, IHttpResponse } from "@/services/http";
+import axios, { InternalAxiosRequestConfig } from "axios";
+import { IHttpClient, IHttpResponse, IHttpClientRequestParams } from "@/services/http";
 import { createQueryString } from "@/utils/queryString";
 
-const { NEXT_PUBLIC_API_URL, NEXT_PUBLIC_COOKIE_NAME } = process.env
+const { NEXT_PUBLIC_API_URL, NEXT_PUBLIC_COOKIE_NAME } = process.env;
 
 const client = axios.create({
   baseURL: NEXT_PUBLIC_API_URL,
-  paramsSerializer: createQueryString,
+  paramsSerializer: {
+    serialize: createQueryString,
+  },
   timeout: 7200000,
-})
+});
 
-client.interceptors.request.use((config: AxiosRequestConfig) => {
-  const token: string | undefined = cookies.get(NEXT_PUBLIC_COOKIE_NAME as string);
-  if (token != null) config.headers.Authorization = `Bearer ${token}`;
+client.interceptors.request.use(
+  (config: InternalAxiosRequestConfig): InternalAxiosRequestConfig => {
+    const token: string | undefined = cookies.get(
+      NEXT_PUBLIC_COOKIE_NAME as string
+    );
+    if (token != null) config.headers.Authorization = `Bearer ${token}`;
 
-  return config;
-})
+    return config;
+  }
+);
 
 export class HttpClientAxios implements IHttpClient {
-  async request<T, R>(
-    url: string,
-    method: HttpMethod,
-    params: T
-  ): Promise<IHttpResponse<R>> {
-    const response = await client({ url, method, params });
+  async request<Response, Payload = undefined, Params = undefined>({
+    url,
+    method,
+    params,
+    payload,
+    baseURL,
+  }: IHttpClientRequestParams<Payload, Params>): Promise<
+    IHttpResponse<Response>
+  > {
+    const sanitizedParams = 
```

Comment:
It would be interesting to create a pure function to sanitize the parameters

Reply:
I wrote a new function called sanitizeParams, but I kept the variable sanitized params to keep the code readable.